node/src/Pos/Block/Logic/Creation.hs

```diff
@@ -138,23 +142,12 @@ createGenesisBlockDo
     -> HeaderHash
     -> m (HeaderHash, Maybe (GenesisBlock ssc))
 createGenesisBlockDo epoch leaders tip = do
-    let noHeaderMsg =
-            "There is no header is DB corresponding to tip from semaphore"
-    tipHeader <- maybeThrow (DBMalformed noHeaderMsg) =<< DB.blkGetHeader tip
+    tipHeader <- DB.blkGetHeaderOrThrow tip
```

Comment:
Actually there is `getTipHeader` in `Pos.DB.DB`.

Reply:
Isn't it better to use the tip provided by the lock here?

Reply:
No, tip provided by the lock should never be used and will be removed. It's not guaranteed to be the same as the one in DB and DB is the right source of truth.

Even if later we introduce a way to safely store tip in-memory, I still find it better to have a helper which reads this tip from some mutable variable and fetches header from DB. But now it's not the case, so just use `DB.getTipHeader`.

Reply:
Alright, I changed the code so it doesn't use the tip from the lock.